Modules/Module_Import/Import/xASL_imp_DetermineSubjectStructure.m

```diff
@@ -153,10 +148,19 @@
         warning('x.SUBJECTS is undefined, data loading is going to fail...');
     end
     
-
+	% If tokenVisitAliases is not provided, we assume visits are not renamed and create a dummy tokenVisitAliases list
+	if ~isfield(x.modules.import.imPar, 'tokenVisitAliases') || isempty(x.modules.import.imPar.tokenVisitAliases)
+		% Create a unique list of sessions
+		listSessions = unique(listSessions);
```

Comment:
Calling it sessions here is tricky as we call runs sessions in these import functions (per legacy nomenclature). We need to keep legacy nomenclature throughout to avoid confusion and bugs.

So should become listVisits

Reply:
You know that this is your code that I just modified? ;)

So the term "SubjectSession" is something you have mostly everywhere in the code. And you even have that in x-structure. I'm fine changing it here, but we have x.SUBJECTS and x.SESSIONS everywhere in the entire xASL for as long as xASL exists.

I agree that this is confusing, but that's a larger problem than this single function.

Reply:
Yes and SubjectSession x.SESSIONS all through the old ExploreASL code, session was run.

In this code, you are tweaking visits. So if you tweak something that was called session, you might change the session/run behavior rather than the visit/session behavior.

This is actually the whole reason of course why we should use one nomenclature throughout the whole import.

Reply:
1.Yes, I'm sure the 'ASL_' should be removed.

2. I didn't tweak anything here - this function was already using session as we are using legacy code to import into BIDS - there's not a perfect unique solution. I changed everything to visits here, but it's playing with unimportant details with no impact.